core/trino-main/src/main/java/io/trino/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -2568,22 +2559,33 @@ private Type coerceToSingleType(StackableAstVisitorContext<Context> context, Str
         {
             // determine super type
             Type superType = UNKNOWN;
+
+            // Use LinkedHashMultimap to preserve order in which expressions are analyzed within IN list
```

Comment:
Analysis order shouldn't matter, does it?

Reply:
It does. see `testInvalidTypeArray`

Reply:
Result will be the same but exception should be raised on first non-coercible value